vllm/model_executor/layers/fused_moe/modular_kernel.py

```diff
@@ -716,20 +719,30 @@ def __init__(
         fused_experts: FusedMoEPermuteExpertsUnpermute,
         shared_experts: torch.nn.Module | None = None,
         shared_experts_stream: torch.cuda.Stream | None = None,
-        parallel_config: ParallelConfig | None = None,
+        moe_parallel_config: FusedMoEParallelConfig | None = None,
     ):
         super().__init__()
         self.prepare_finalize = prepare_finalize
         self.fused_experts = fused_experts
         self.shared_experts = shared_experts
         self.shared_experts_stream = shared_experts_stream
 
-        # cache whether this worker is using DP+EP
-        if parallel_config is None:
-            parallel_config = get_current_vllm_config().parallel_config
+        # prefer an explicit FusedMoEParallelConfig when available
+        # (from FusedMoE layers / tests), otherwise derive one from the
+        # current vLLM config.
+        if moe_parallel_config is None:
+            vllm_config = get_current_vllm_config()
+            parallel_config = vllm_config.parallel_config
+            moe_parallel_config = FusedMoEParallelConfig.make(
+                tp_size_=parallel_config.tensor_parallel_size,
+                pcp_size_=parallel_config.pipeline_parallel_size,
+                dp_size_=parallel_config.data_parallel_size,
+                vllm_parallel_config=parallel_config,
+            )
```

Comment:
I think we should just assume that this is a non-EP case if no `moe_parallel_config` is passed.  It will avoid the call to `get_current_vllm_config` which is causing all the warnings/logs.